frontend/amundsen_application/api/utils/metadata_utils.py

```diff
@@ -128,6 +134,8 @@ def marshall_table_full(table_dict: Dict) -> Dict:
     for col in columns:
         # Set editable state
         col['is_editable'] = is_editable
+        if col['type_metadata']:
```

Comment:
the other functions below this do the if-not-none check themselves, so put that into _set_type_metadata_is_editable